src/datatypes/src/data_type.rs

```diff
@@ -183,6 +183,12 @@ impl ConcreteDataType {
     }
 }
 
+impl From<&ConcreteDataType> for ConcreteDataType {
```

Comment:
It's a little wired to see a reference "into" a concrete value. Should have just used `clone`.

Reply:
It's convenient to use in snafu, simplifies the code from
```rust
TestSnafu {
   datatype: datatype.clone(),
}.fail()
```

to:

```rust
TestSnafu {
   datatype,
}.fail()
```